### PR TITLE
Allow setting a different documentation repository

### PR DESCRIPTION
**Pull request by bpolaszek**
This feature enables setting a different documentation repository with an optional environment variable, for development purposes.

Usage: 

```bash
$ DOCUMENTATION_REPOSITORY=https://github.com/bpolaszek/api-platform-docs ./bin/retrieve-documentation
```

### CODE REVIEW

File: bin/checkout-documentation [hunk 1]
```diff
@@ -3,14 +3,14 @@
 const {execSync} = require('child_process');
 const fs = require('fs');
 const path = require('path');
-const {docPagesDirectory, current, versions} = require('../constants');
+const {docPagesDirectory, current, versions, documentationRepository} = require('../constants');
```
Comment (lex111) [inline bin/checkout-documentation]:
```suggestion
const {docPagesDirectory, current, versions, docRepository} = require('../constants');
```

File: bin/checkout-documentation [hunk 2]
```diff
@@ -3,14 +3,14 @@
...
 const versionHelper = require('../src/lib/versionHelper');
 
 console.info('\x1b[36m', `Deleting ${docPagesDirectory}`, '\x1b[37m');
 execSync(`rm -rf ${docPagesDirectory}`);
 
 console.info('\x1b[36m', `Cloning the current version in ${docPagesDirectory}${current}`, '\x1b[37m');
-execSync(`git clone https://github.com/api-platform/docs.git ${docPagesDirectory}${current}`);
+execSync(`git clone ${documentationRepository } ${docPagesDirectory}${current}`);
```
Comment (lex111) [inline bin/checkout-documentation]:
```suggestion
execSync(`git clone ${docRepository} ${docPagesDirectory}${current}`);
```

File: constants.js [hunk 3]
```diff
@@ -5,4 +5,5 @@ module.exports = Object.freeze({
   currentVersion: '2.5',
   masterVersion: '2.6',
   siteUrl: 'https://api-platform.com',
+  documentationRepository: process.env.DOCUMENTATION_REPOSITORY || 'https://github.com/api-platform/docs.git',
```
Comment (lex111) [inline constants.js]:
```suggestion
  docRepository: process.env.DOCUMENTATION_REPOSITORY || 'https://github.com/api-platform/docs.git',
```

**Comment by bpolaszek** [comment]
> Code style fixes.

Do you think the env var should be renamed `DOC_REPOSITORY` as well?

**Comment by lex111** [comment]
@bpolaszek yep, but it’s even better to create a separate `.env` file by analogy with [`.env.gatsby`](https://github.com/api-platform/website/blob/master/.env.gatsby) (it is used only for the site), and the new `.env` file will have environment variables that do not belong to Gatsby website. It seems logical?

File: constants.js [hunk 4]
```diff
@@ -5,4 +5,5 @@ module.exports = Object.freeze({
   currentVersion: '2.5',
   masterVersion: '2.6',
   siteUrl: 'https://api-platform.com',
+  docRepository: process.env.DOC_REPOSITORY,
```
Comment (lex111) [inline constants.js]: Сan this change be reverted?

File: bin/checkout-documentation [hunk 5]
```diff
@@ -1,16 +1,20 @@
...
 
+const docRepository = process.env.DOC_REPOSITORY;
+console.info('\x1b[36m', `Using repository: ${docRepository}`, '\x1b[37m');
+
 console.info('\x1b[36m', `Deleting ${docPagesDirectory}`, '\x1b[37m');
 execSync(`rm -rf ${docPagesDirectory}`);
 
 console.info('\x1b[36m', `Cloning the current version in ${docPagesDirectory}${current}`, '\x1b[37m');
-execSync(`git clone https://github.com/api-platform/docs.git ${docPagesDirectory}${current}`);
+execSync(`git clone ${docRepository } ${docPagesDirectory}${current}`);
```
Comment (lex111) [inline bin/checkout-documentation]:
```suggestion
execSync(`git clone ${docRepository} ${docPagesDirectory}${current}`);
```

Comment (bpolaszek) [reply bin/checkout-documentation]: Good catch.

File: bin/checkout-documentation [hunk 6]
```diff
@@ -1,16 +1,20 @@
 #!/usr/bin/env node
 
+require('dotenv').config();
 const {execSync} = require('child_process');
 const fs = require('fs');
 const path = require('path');
 const {docPagesDirectory, current, versions} = require('../constants');
```
Comment (lex111) [inline bin/checkout-documentation]:
```suggestion
const {docPagesDirectory, current, versions} = require('../constants');
const docRepository = process.env.DOC_REPOSITORY;
```

File: bin/checkout-documentation [hunk 7]
```diff
@@ -1,16 +1,20 @@
...
 const versionHelper = require('../src/lib/versionHelper');
 
+const docRepository = process.env.DOC_REPOSITORY;
```
Comment (lex111) [inline bin/checkout-documentation]:
```suggestion
```

**Comment by bpolaszek** [comment]
> Last fix to keep consistency let's do this.

That's what I did at first, but my IDE warns me about this:
> ESLint: Expected 1 empty line after require statement not followed by another require (import/newline-after-import)

**Comment by lex111** [comment]
Ah, I see, all right.

File: .env [hunk 8]
```diff
@@ -0,0 +1 @@
+DOC_REPOSITORY=https://github.com/api-platform/docs.git
```
Comment (dunglas) [inline .env]:
```suggestion
DOCS_REPOSITORY=https://github.com/api-platform/docs.git
```

**Comment by dunglas** [comment]
Thanks @bpolaszek

**Pull request merged**